sha2/src/aarch64.rs

```diff
@@ -0,0 +1,11 @@
+// From sys/auxv.h
+const AT_HWCAP: u64 = 16;
+const HWCAP_SHA2: u64 = 64;
+extern {
+    fn getauxval(type_: u64) -> u64;
```

Comment:
Are you sure we don't need any linker options to use `getauxval`? It may work only because `std` links it as well, so for `no_std` app things could break.

Reply:
`no_std` only means we don’t use the `std` crate, not that we don’t use the system libc, right?  Do you have a Linux system where this isn’t the case?

Reply:
But you still have to link to `libc`. I think the best approach here will be to use [`libc::getauxval`](https://docs.rs/libc/0.2.66/libc/fn.getauxval.html). Of course `libc` should be an optional dependency, and we probably should gate your assembly not only on the `asm` feature, but also on the new `libc` feature (or introduce an alias for it).

Reply:
https://github.com/rust-lang/cargo/issues/1197 is preventing the obvious solution to enable this feature conditionally, I guess I’ll have to fix it first. :/

Reply:
I think for now we can use something like this:
```toml
[dependencies]
libc = { version = "0.2", optional = true }
sha1-asm = { version = "0.4", optional = true }

[features]
asm = ["asm-hashes"]
# has effect only for AArch64 Linux system
aarch64-crypto = ["asm", "libc"]
```

Can you open an issue in the `libc` repo regarding those problems?

Reply:
I made a PR: https://github.com/rust-lang/libc/pull/1638